app/build.gradle

```diff
@@ -29,22 +31,55 @@ android {
         versionCode versionMajor * 10000 + versionMinor * 1000 + versionPatch * 100 + versionBuild
         versionName "${versionMajor}.${versionMinor}.${versionPatch}"
 
-        resValue "string", "github_secret", pockethub_github_secret
-        resValue "string", "github_client", pockethub_github_client
+        if (project.hasProperty('pockethub_github_client')) {
+            resValue "string", "github_client", pockethub_github_client
+            clientIdIsNotSet = false
+        } else {
+            resValue "string", "github_client", "dummy_client"
+        }
+
+        if (project.hasProperty('pockethub_github_secret')) {
+            resValue "string", "github_secret", pockethub_github_secret
+            clientSecretIsNotSet = false
+        } else {
+            resValue "string", "github_secret", "dummy_secret"
+        }
 
-        if("dummy_secret".equals(pockethub_github_secret) || "dummy_client".equals(pockethub_github_client)){
-            logger.warn("You won't be able to login, because the oauth client and secret aren't set")
+        if (clientIdIsNotSet || clientSecretIsNotSet) {
+            logger.warn("You won't be able to login, because the oauth client or secret isn't set")
             logger.warn("Read the README.md: https://github.com/pockethub/PocketHub#setup-environment")
         }
 
-        String oauth = pockethub_github_callback
+        if (project.hasProperty('pockethub_bugsnag_api_key')) {
+            resValue "string", "bugsnag_api_key", pockethub_bugsnag_api_key
+        } else {
+            resValue "string", "bugsnag_api_key", "bugsnag-api-key"
+        }
+
+        String oauth = ""
+        if (project.hasProperty('pockethub_github_callback')) {
+            oauth = pockethub_github_callback
+        } else {
+            oauth = "http://dummy.example.com"
+        }
+
         resValue "string", "github_oauth", oauth
-        resValue "string", "github_oauth_scheme", oauth != "DEFAULT" ? oauth.split("://")[0] : "DEFAULT"
+        resValue "string", "github_oauth_scheme", oauth != "http://dummy.example.com" ? oauth.split("://")[0] : "http://dummy.example.com"
```

Comment:
This is now needed, we can use only <code> resValue "string", "github_oauth_scheme", oauth.split("://")[0]</code>